src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -818,20 +819,46 @@ public RequestLineStatus TakeStartLine(SocketInput input)
                     }
                 }
 
-                string method;
-                var begin = scan;
-                if (!begin.GetKnownMethod(out method))
+                end.Take();
+
+                var methodStart = scan;
+                var methodEnd = methodStart;
+                if (methodEnd.Seek(ref _vectorSpaces, ref end) == -1)
```

Comment:
It must be faster to call GetKnownMethod first in the normal case when there's a valid request with a known method right?